File: Raven.Client.Lightweight/Document/ResponseTimeInformation.cs
// -----------------------------------------------------------------------
//  <copyright file="ResponseTimeInformation.cs" company="Hibernating Rhinos LTD">
//      Copyright (c) Hibernating Rhinos LTD. All rights reserved.
//  </copyright>
// -----------------------------------------------------------------------
using System;
using System.Collections.Generic;
using System.Linq;

namespace Raven.Client.Document
{
    public class ResponseTimeInformation
    {

        public ResponseTimeInformation()
        {
            DurationBreakdown = new List<ResponseTimeItem>();
            TotalServerDuration = new TimeSpan();
            TotalClientDuration = new TimeSpan();
        }
        public TimeSpan TotalServerDuration { get; set; }
        public TimeSpan TotalClientDuration { get; set; }

        public List<ResponseTimeItem> DurationBreakdown { get; set; }

        internal void ComputeServerTotal()
        {
            TotalServerDuration = new TimeSpan(DurationBreakdown.Sum(x => x.Duration.Ticks));
        }
    }

    public class ResponseTimeItem
    {
        public string Url { get; set; }
        public TimeSpan Duration { get; set; }
    }
}

File: Raven.Client.Java/src/main/java/net/ravendb/client/document/ResponseTimeInformation.java
package net.ravendb.client.document;

import java.util.ArrayList;
import java.util.List;


public class ResponseTimeInformation {

  private Long totalServerDuration;
  private Long totalClientDuration;

  private List<ResponseTimeItem> durationBreakdown;


  public ResponseTimeInformation() {
    totalClientDuration = 0L;
    totalServerDuration = 0L;
    durationBreakdown = new ArrayList<>();
  }

  public Long getTotalServerDuration() {
    return totalServerDuration;
  }


  public void setTotalServerDuration(Long totalServerDuration) {
    this.totalServerDuration = totalServerDuration;
  }


  public Long getTotalClientDuration() {
    return totalClientDuration;
  }


  public void setTotalClientDuration(Long totalClientDuration) {
    this.totalClientDuration = totalClientDuration;
  }


  public List<ResponseTimeItem> getDurationBreakdown() {
    return durationBreakdown;
  }


  public void setDurationBreakdown(List<ResponseTimeItem> durationBreakdown) {
    this.durationBreakdown = durationBreakdown;
  }

  public void computeServerTotal() {
    long total = 0;
    for (ResponseTimeItem item : durationBreakdown) {
      total += item.getDuration();
    }
    totalServerDuration = total;
  }

  public static class ResponseTimeItem {
    private String url;
    private Long duration;

    public String getUrl() {
      return url;
    }

    public void setUrl(String url) {
      this.url = url;
    }

    public Long getDuration() {
      return duration;
    }

    public void setDuration(Long duration) {
      this.duration = duration;
    }

  }
}
